src/Compilers/CSharp/Test/Symbol/Symbols/OverriddenOrHiddenMembersTests.cs

```diff
@@ -3983,5 +3983,441 @@ public class Required : ValidatorBase<object>
+        }
+
+        [Fact]
+        [CompilerTrait(CompilerFeature.ReadonlyReferences)]
+        public void MethodOverloadsShouldPreserveReadOnlyRefnessInReturnTypes()
+        {
+            var text = @"
+abstract class BaseClass
+{
+    protected int x = 0 ;
+    public virtual ref int Method1() { return ref x; }
+    public virtual ref readonly int Method2() { return ref x; }
+}
+class ChildClass : BaseClass
+{
+    public override ref readonly int Method1() { return ref x; }
+    public override ref int Method2() { return ref x; }
+}";
+
+            var comp = CreateCompilationWithMscorlib(text).VerifyDiagnostics(
+                // (11,29): error CS8148: Reference signature of the return type of 'ChildClass.Method2()' does not match overridden member 'BaseClass.Method2()'
+                //     public override ref int Method2() { return ref x; }
+                Diagnostic(ErrorCode.ERR_CantChangeRefnessOfReturnOnOverride, "Method2").WithArguments("ChildClass.Method2()", "BaseClass.Method2()").WithLocation(11, 29),
+                // (10,38): error CS8148: Reference signature of the return type of 'ChildClass.Method1()' does not match overridden member 'BaseClass.Method1()'
+                //     public override ref readonly int Method1() { return ref x; }
+                Diagnostic(ErrorCode.ERR_CantChangeRefnessOfReturnOnOverride, "Method1").WithArguments("ChildClass.Method1()", "BaseClass.Method1()").WithLocation(10, 38));
+        }
+
+        [Fact]
+        [CompilerTrait(CompilerFeature.ReadonlyReferences)]
+        public void PropertyOverloadsShouldPreserveReadOnlyRefnessInReturnTypes()
+        {
+            var code = @"
+class A
+{
+    protected int x = 0;
+    public virtual ref int Property1 { get { return ref x; } }
+    public virtual ref readonly int Property2 { get { return ref x; } }
+}
+class B : A
+{
+    public override ref readonly int Property1 { get { return ref x; } }
+    public override ref int Property2 { get { return ref x; } }
+}";
+
+            var comp = CreateCompilationWithMscorlib(code).VerifyDiagnostics(
+                // (11,29): error CS8148: Reference signature of the return type of 'B.Property2' does not match overridden member 'A.Property2'
+                //     public override ref int Property2 { get { return ref x; } }
+                Diagnostic(ErrorCode.ERR_CantChangeRefnessOfReturnOnOverride, "Property2").WithArguments("B.Property2", "A.Property2").WithLocation(11, 29),
+                // (10,38): error CS8148: Reference signature of the return type of 'B.Property1' does not match overridden member 'A.Property1'
+                //     public override ref readonly int Property1 { get { return ref x; } }
+                Diagnostic(ErrorCode.ERR_CantChangeRefnessOfReturnOnOverride, "Property1").WithArguments("B.Property1", "A.Property1").WithLocation(10, 38));
+        }
+    }
```

Comment:
Are `ref readonly` parameters supported for indexers? `object this[ref readonly int i] { get; set; }`